source/common/http/filter_manager.cc

```diff
@@ -182,12 +182,7 @@ bool ActiveStreamFilterBase::commonHandleAfterDataCallback(FilterDataStatus stat
         status == FilterDataStatus::StopIterationAndWatermark) {
       buffer_was_streaming = status == FilterDataStatus::StopIterationAndWatermark;
       commonHandleBufferData(provided_data);
-    } else if (complete() && !hasTrailers() && !bufferedData()) {
-      if (parent_.state_.destroyed_) {
-        // If the stream is destroyed, no need to handle the data buffer or trailers.
-        // This can occur if the filter calls sendLocalReply.
```

Comment:
I think this comment is useful? Add it back?

/wait

Reply:
Done. Not really sure where to place it since the if is gone.